### PR TITLE
[IMP] sale: clean SO & SOL fields

### PR DESCRIPTION
**Pull request by Feyensv**
* order by logic/dependencies
  * reduce stacktrace depth during create precomputation
  by computing the computed fields dependencies before the field itself
  not during the field computation (if the field was defined after)
* guidelines applications
  * single quotes for tech strings
  * double quotes for strings shown to user
  * respect max line length
  * try to follow consistent order for field attributes

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/models/sale_order_line.py [hunk 1]
```diff
@@ -5,17 +5,247 @@
...
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string="Product",
+        change_default=True, ondelete='restrict', check_company=True,
+        domain="[('sale_ok', '=', True), '|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    product_template_id = fields.Many2one(
+        string="Product Template",
+        related='product_id.product_tmpl_id',
+        domain=[('sale_ok', '=', True)])
+    product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
```
Comment (fmdl) [inline addons/sale/models/sale_order_line.py:76]: add `depends=['product_id']` to prevent big triger @rco-odoo

Comment (Feyensv) [reply addons/sale/models/sale_order_line.py:76]: Useless, the field is not stored and only computed when the record is shown.

File: addons/sale/models/sale_order.py [hunk 2]
```diff
@@ -12,25 +12,264 @@
 
 
 class SaleOrder(models.Model):
-    _name = "sale.order"
+    _name = 'sale.order'
```
Comment (MiquelRForgeFlow) [inline addons/sale/models/sale_order.py:15]: why?

Comment (Feyensv) [reply addons/sale/models/sale_order.py:15]: For our team guideline saying that user-shown strings are defined in double quotes, and tech strings are specified as single quotes :D.

**Review by Feyensv — COMMENTED** [review]
robodoo r+ rebase-ff

**Pull request closed without merging**